docs/spark-writes.md

```diff
@@ -339,74 +331,55 @@ USING iceberg
-import org.apache.spark.sql.types.DataTypes
-
-IcebergSpark.registerBucketUDF(spark, "iceberg_bucket16", DataTypes.LongType, 16)
-```
-
-{{< hint info >}}
-Explicit registration of the function is necessary because Spark doesn't allow Iceberg to provide functions.
-[SPARK-27658](https://issues.apache.org/jira/browse/SPARK-27658) is filed to enable Iceberg to provide functions
-which can be used in query.
-{{< /hint >}}
-
-Here we just registered the bucket function as `iceberg_bucket16`, which can be used in sort clause.
-
-If you're inserting data with SQL statement, you can use the function like below:
-
-```sql
-INSERT INTO prod.db.sample
-SELECT id, data, category, ts FROM another_table
-ORDER BY iceberg_bucket16(id)
-```
-
-If you're inserting data with DataFrame, you can use the function like below:
-
-```scala
-data.sortWithinPartitions(expr("iceberg_bucket16(id)"))
-    .writeTo("prod.db.sample")
-    .append()
-```
 
+There are 3 options for `write.distribution-mode`
+
+* `none` - This is the previous default for Iceberg.<p> This mode does not require any shuffles or sort to be performed
+automatically by Spark. Because no work is done automatically by Spark, the data must be either locally or globally 
+sorted manually by partition value. To reduce the number of files produced during writing, using a global sort is recommended.<p> 
+A local sort can be avoided by using the Spark [write fanout](#write-properties) property but this will cause all file handles to 
+remain open until each write task has completed. 
+* `hash` - This mode is the new default and requests that Spark uses a hash-based exchange to shuffle the incoming
+write data before writing. Practically, this means that each row is hashed based on the row's partition value and then placed
+in a corresponding Spark task based upon that value. Further division and coalescing of tasks may take place based on 
+the [Spark's Adaptive Query planning](#controlling-file-sizes).
+* `range` - This mode requests that Spark perform a range based exchanged to shuffle the data before writing. This is
+a two stage procedure which is more expensive than the `hash` mode. The first stage samples the data to be written based
+on the partition and sort columns, this information is then used in the second stage to shuffle data into tasks. Each
+task gets an exclusive range of the input data which clusters the data by partition and also globally sorts it.
+While this is more expensive than the hash distribution, the global ordering can be beneficial for read performance if
+sorted columns are used during queries. Further division and coalescing of tasks may take place based on
+  the [Spark's Adaptive Query planning](#controlling-file-sizes).
+
+
+## Controlling File Sizes
+
+When writing data to Iceberg with Spark, it's important to note that Spark cannot write a file larger than a Spark 
+task. This means although Iceberg will always roll over a file when it grows to 
```

Comment:
This is a great section.  While we are at it, would it also help new users to explicitly mention partitions, ie, 

`it's important to note that Spark cannot write a file larger than a Spark task, and files cannot span across Iceberg partitions`